src/App.js

```diff
@@ -2,24 +2,99 @@ import React from 'react';
 
 import './App.scss';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
+class App extends React.Component {
+  timerID = null;
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      date: new Date(),
+      isClockVisible: true,
+      clockName: 'clock',
+    };
+  }
```

Comment:
```suggestion
    this.state = {
      date: new Date(),
      isClockVisible: true,
      clockName: 'clock',
    };
```
No need in constructor here. Just use `state` object